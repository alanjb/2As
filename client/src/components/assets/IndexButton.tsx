import React from 'react';
import indexButton from '../../assets/index-button.svg';

const IndexButton = (Props: any) => {
  const {toggleIndexSidePanel} = Props;
  return (
    <React.Fragment>
      <header className="PiXIcon">
        <div className="PiXIcon-Component">
          <img className="index-button" src={indexButton} onClick={toggleIndexSidePanel} alt="index"/>
        </div>
      </header>
    </React.Fragment>
  )
}

type MyProps = {
  toggleIndexSidePanel: () => any;
};

type Props = MyProps;

export default IndexButton;