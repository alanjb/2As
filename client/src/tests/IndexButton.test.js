import React from 'react';
import IndexButton from '../components/assets/IndexButton';
import Enzyme, { shallow, mount, render} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter });

/**
 * Initialize common properties to be passed
 * @param {*} props properies to be override
 */
function createTestProps(props) {
    return {
        ...props,
    };
}

let component, props;

beforeEach(() => {
    props = createTestProps();
    component = shallow(<IndexButton {...props} /> );
});

describe("IndexButton Component", () => {
    it('renders correctly in DOM', () => {
        shallow(
            <IndexButton text="Test" />
        );
    });

    it('expects to find an img HTML element in the DOM that represents the index button icon button and render it', () => {
        expect(component.find('img')).toHaveLength(1);
    });

    it('should be defined', () => {
        expect(IndexButton).toBeDefined();
    });

    it('should render a hamburger button', () => {
        const indexbuttonicon = component.find('.index-button');
        expect(indexbuttonicon.length).toBe(1);
        expect(component).toMatchSnapshot();
        // expect(mockToggleIndexSidePanelCallBack.mock.calls.length).toEqual(1);
    });

    it('should call the onClick function when index button is clicked', () => {
        const mockToggleIndexSidePanelCallBack = jest.fn(); 
        component.instance().toggleIndexSidePanel = mockToggleIndexSidePanelCallBack;
        component.find()
        expect(mockToggleIndexSidePanelCallBack).toHaveBeenCalled(1);
        const indexbuttonicon = component.find('.index-button');
        indexbuttonicon.first().props().onClick();
    });
});