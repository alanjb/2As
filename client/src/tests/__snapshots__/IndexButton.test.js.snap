// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexButton Component should render a hamburger button 1`] = `
<Fragment>
  <div
    className="IndexButton-Component"
  >
    <img
      alt="index"
      className="index-button"
      src="index2svg-01.svg"
    />
  </div>
</Fragment>
`;
